### ISSUE TITLE
Windows 10 taskbar doesn't work anymore on build 348

### ISSUE THREAD

**Issue by frankbons** [valinet/ExplorerPatcher#440]
Windows 10 taskbar doesn't work anymore on build 348. 
I hope it will work again in the next release. I miss this great patcher!

**Comment by SpIdErPiGgY1984** [comment]
Same here, after the new Win 11 beta update, this patch broke

**Comment by udeshplus** [comment]
Same for me :(  

```
Edition	Windows 11 Pro
Version	21H2
Installed on	‎19/‎11/‎2021
OS build	22000.348
Experience	Windows Feature Experience Pack 1000.22000.348.0
```

**Comment by hk1956** [comment]
same here,   in 22000.348, Windows File Explorer repeatedly restarts so fast that it cannot open any windows.
![image](https://user-images.githubusercontent.com/94793842/142767542-a5cdd902-9983-4c70-aa92-9ab1695428d9.png)

**Comment by frankbons** [comment]
> same here, in 22000.348, Windows File Explorer repeatedly restarts so fast that it cannot open any windows. ![image](https://user-images.githubusercontent.com/94793842/142767542-a5cdd902-9983-4c70-aa92-9ab1695428d9.png)

I can reproduce this when switching to Windows 10 switcher. On windows 11 switcher the restart-loop stops.

**Comment by BlackRockSoul** [comment]
Why closed since it doesn't work? W10 switcher is one of the best features it has because W11 switcher lags so hard for me for some reason (MS is the reason).
So anyway how do I turn it off if it doesn't even start?

**Comment by frankbons** [comment]
I am sorry. I didn't mean to close this issue. Pushed the wrong button.

**Comment by frankbons** [comment]
"UPD. Fixed somehow, changed the OSBuild version to 10.0.22000.348 by Computer\HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\ExplorerPatcher
Totally not sure it's the right way to fix but works for me."

Wow. This works for me too. Thanks!

**Comment by udeshplus** [comment]
@BlackRockSoul @frankbons Can you please tell me how to change the OSBuild version from the given registry address? This is what I can see in the given location. 
![image](https://user-images.githubusercontent.com/30784707/142776894-d016af30-93c2-4bb8-9dc2-b3962306cda6.png)

**Comment by frankbons** [comment]
> @BlackRockSoul @frankbons Can you please tell me how to change the OSBuild version from the given registry address? This is what I can see in the given location. ![image](https://user-images.githubusercontent.com/30784707/142776894-d016af30-93c2-4bb8-9dc2-b3962306cda6.png)

That is strange. This is what I see:
![image](https://user-images.githubusercontent.com/92643075/142777131-b3f542b8-cef1-4085-a1ef-36bd6ab8b57a.png)

**Comment by SpIdErPiGgY1984** [comment]
All I see is:

![2021-11-21 21 07 30](https://user-images.githubusercontent.com/7800936/142777264-6773ca8f-6946-4cdf-9fcc-607e340e7da2.jpg)

**Comment by udeshplus** [comment]
@frankbons Thanks! I will add a new multi-string value.

**Comment by SpIdErPiGgY1984** [comment]
Adding OSBuild with 10.0.22000.348 alsi did not work for me


![2021-11-21 21 24 03](https://user-images.githubusercontent.com/7800936/142777684-56c57ea7-7a7d-49b0-bfbc-bd51a7f2e60b.jpg)

**Comment by frankbons** [comment]
> 



> @frankbons Thanks! I will add a new string value and see.
> 
> PS: This did not fix the issue for me.

My explorer patcher version is currently 22000.318.38.2. This is working for me. Maybe you should try this?

**Comment by udeshplus** [comment]
@frankbons Maybe that is the case. Where can I find this version? Can you share the link for this version?

**Comment by SpIdErPiGgY1984** [comment]
> > 
> 
> > @frankbons Thanks! I will add a new string value and see.
> > PS: This did not fix the issue for me.
> 
> My explorer patcher version is currently 22000.318.38.2. This is working for me. Maybe you should try this?

Nope, just tried, same issue

**Comment by frankbons** [comment]
I seem not to be able to share the file dxgi.dll that is in my C:\Windows. The forum will not let me upload it. Any ideas how I can do that?

**Comment by frankbons** [comment]
> > > 
> > 
> > 
> > > @frankbons Thanks! I will add a new string value and see.
> > > PS: This did not fix the issue for me.
> > 
> > 
> > My explorer patcher version is currently 22000.318.38.2. This is working for me. Maybe you should try this?
> 
> Nope, just tried, same issue

I don't know what could be the difference between us. I did nothing special ... All I did was a update from settings (from explorer patcher). I then went to 38.2 all by it self

**Comment by SpIdErPiGgY1984** [comment]
I uninstalled with Revo uninstaller, deleted files en registry... Reinstalled this patcher, still no luck

**Comment by frankbons** [comment]
> I uninstalled with Revo uninstaller, deleted files en registry... Reinstalled this patcher, still no luck

I am sorry. I don't have a clue why it works for me, and not for you guys.
![image](https://user-images.githubusercontent.com/92643075/142781010-034f6c57-4ffa-4f93-a7c8-04f32049005d.png)

**Comment by SpIdErPiGgY1984** [comment]
Too bad 😒 Really love and need this patch. Hope a fix will be found soon...
Btw, I'm also Dutch 🤣

**Comment by valinet** [comment]
The proper fix is to stay on stable builds, imo. I explained multiple times why this happens: Microsoft apparently does not publish symbols for beta/release preview builds. I don’t know why. They just don’t do it. Symbols are required to enable some functionality which cannot be enabled without or for which there is currently a lack of a reimplementation in the source of EP itself. It’s hard to get rid of this need 100% anyway, and why do it… they help achieve some cool stuff, and it’s mostly this issue on some builds that is still an issue on Microsoft’s part, not mine, really. Symbols offer a mapping between human friendly names and addresses in the executable, so it helps identify where functions doing certain stuff are in an executable.

All this hacks proposed here and there are just hacks. Sharing them around without offering a technical explanation only has the potential of causing me and this project problems: while addresses from previous builds might work, it is not guaranteed to, or rather unverified. It’s just a coincidence. I cannot ship coincidences in official builds. People use these hacks, run it, some random shit happens and they start opening tickets and blaming EP without being able to understand exactly what may cause it. It’s a nightmare to debug just because you have to run beta builds for an OS you hate. I hardly understand it, but whatever. 

For the past 2 builds, I have manually patched in the symbols, by identifying the addresses in the new DLL manually. It’s a tedious and tiring process only the human brain is able to comprehend and perform correctly; it’s repetitive work which I cannot keep pace with for each build they release every couple of days. Computers are dumb and pretty poor when there isn’t an exact algorithm to follow. Identifying functions in a new build by comparing it to an older build is like driving a car. Various dudes tell you it’s doable with conventional computing, while no one in the next 100 years is willing to assume responsibility and say the car should be found guilty (or the automaker by extension) in case of an accident, instead of the driver, when self driving. Because that is all it is about, that is what people are really expecting, it’s not something that will happen too soon. Same hear, computers are dumb enough to find difficult this tedious but relatively easy job for a human brain. But what’s the expectation, patch in addresses manually for the next 2 years? I simply don’t want to do that. And I also don’t have time to necessarily work on this every time Microsoft releases some new build. I expect the community to step in, understand the problem and try to come up with a real solution, not a patch on top of a patch for a mostly self created issue.

The things that currently need symbols are:

* Win-X menu is taken from twinui.pcshell.dll. A free and open reimplementation could be made, but it takes some effort.
* Skinned menus are taken from twinui.pcshell.dll. A free and open reimplementation could be made, but it takes some effort.
* All Start menu related functionality. Kind of impossible to do without symbols. Although not totally impossible.

Another solution could always be, theoretically, to load the stuff from twinui.pcshell.dll from an older version whose addresses we know, we always use that and call it a day, it will always work and those things never change beside being removed altogether eventually, so it’s fine to use it from an older version. The problem with that is redistribution. How does one redistribute or point people to download that file? That and all of this has to be debated, I am not a God, I am willing to learn and share ideas and try to find a solution and I don’t know an answer to anything, I expect to find a way to talk with and debate these things with the community, and get to the best solution regarding this, which I currently do not know, not to open threads upon threads where we discuss how to hack an app that hacks Windows because you upgraded to some random unsupported build that brings nothing in fact, after all.

To mention why some of you had it failing and why it worked for some. The app checks against some hashes and then uses the offsets saved in the registry if the build is similar to what it has recorded in the registry, otherwise it downloads the symbols from Microsoft (or tried to and fails).

When you modify that reg entry to say 10.0.22000.348, you tell the app that symbols for the current build (which is 348) are already downloaded, so it first tries to match against some known hashes, which it fails, and then it loads from the registry. This is where it probably fails for some of you, those that have always run a version with hardcoded addresses, where there was no need to cache the symbols in the registry. Because it doesn’t see any data in there, it yet again tries to download. For others it works, as you have some data there. If it works, it is just a coincidence, no one verified that and as I said, I cannot vouch for things I haven’t verified myself.

So, please, give me your inputs on how to solve this organically. 

Thank you

**Comment by SpIdErPiGgY1984** [comment]
Thank you for the explanation valinet.
I'm gonna be a good boy, and install a "normal" Windows 11... I really need this patch 😁

**Comment by valinet** [comment]
> Thank you for the explanation valinet.
> I'm gonna be a good boy, and install a "normal" Windows 11... I really need this patch 😁

Yeah, it’s not even about being good or bad, it’s about understanding the situation and making an informed decision and deciding how to proceed. A viable solution has to be provided, the approach until now regarding this is not viable long term. I also am trying my best to provide something useful, I feel bad when Microsoft is really to blame here but some people look at how EP doesn’t fix something…

Thank you

**Comment by BraINstinct0** [comment]
Related to this issue, so prob I'll suggest what we can do in the meantime.
There's not much we can do about the delayed/skipped debug symbol uploads, but not none.

I've tried contacting MS and request upload of the pdb files. As expected there's no response. Absolutely zero.

So, while waiting for the upload, anyone willing to help can mail MS about debug symbols not being uploaded fast enough(you don't have to mention ExplorerPatcher as this problem is not just for this repo). [Related page](https://docs.microsoft.com/windows-hardware/drivers/debugger/microsoft-public-symbols)
On their page:
```
Server Status: No known issues ✅
The Microsoft public symbol server is fully operational.
Please report any known issues to windbgfb@microsoft.com.
```
...and that's last edited on 2020.

**Comment by SpIdErPiGgY1984** [comment]
Already fixed here, just re-installed Windows 11.
On build 22000.318, and got this patch back! 😁
No more Beta for me hahaha

**Comment by petrechitashvili** [comment]
> The proper fix is to stay on stable builds

@valinet what's the proper way to stay on the stable build?

Are these settings correct?
![image](https://user-images.githubusercontent.com/15320885/142805559-2390fa77-2cf6-45fb-889e-46d6adc718a9.png)

**Comment by BraINstinct0** [comment]
No, release Preview isn't stable; it's 'almost stable'.
[As MS has announced](https://blogs.windows.com/windows-insider/2021/11/19/releasing-windows-11-build-22000-348-to-beta-and-release-preview-channels/), the current Release Preview is at 22000.348, same as Beta channel.

To be sure, opting out of the Insider Program will work. Further instructions can be easily searched(but be sure to follow the official ones written by MS).

**Comment by BraINstinct0** [comment]
@valinet 
https://docs.microsoft.com/en-us/archive/blogs/windbg/update-on-microsofts-symbol-server
This.
MS already knew that this would be a possible issue at least 5 years ago, and seems to have let the issue persist, so it's highly likely that the debug symbol release delay will happen again and again.
So, not much we can do about the delay, as MS doesn't seem to fix this anytime soon🤔

**Comment by l0oo** [comment]
@valinet

**The proper fix is to stay on stable builds.**

Hello Valinet and thanks for great W11 solution! Real lifesaver!
Well - maybe it is worth to write in bold (and big enough) letters on main page that **EP is for Windows 11 stable builds only**. :)

**Comment by valinet** [comment]
Yeah, will do, I will update the README soon.

**Comment by chip33** [comment]
> Yeah, will do, I will update the README soon.

Hi just an idea but if you still get many issues raised over this a Pinned Issue with "EP only works on Stable builds" would be highly visible on the top of Issues page in case README isn't enough.

**Comment by plofhaan** [comment]
> A viable solution has to be provided, the approach until now regarding this is not viable long term. I also am trying my best to provide something useful, I feel bad when Microsoft is really to blame here but some people look at how EP doesn’t fix something…

Until someone finds a better solution, maybe you could let EP graciously fall back to temporary disabling all the specific functionality that specifically depends on the symbols, each time when it cannot find the symbols? Also maybe, add an option that, when set to enabled, lets the built-in updater function of EP keep automatically retrying to download the symbols at regular time intervals, repeatedly, until the download succeeds? If too many people still keep bothering you about issues next, just let EP send them a toast notification that plays an angry sound. ;)

**Comment by Laserstrahl** [comment]
I had the same issue that ExplorerPatch is broken with the latest windows 11 beta build (22000.348). However I managed to fix it by using the latest build (Build #155) and by editing the registry key
Computer\HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\ExplorerPatcher\OSBuild

from 10.0.22000.318 to 10.0.22000.348

It seems to work. Only bug is, that right-clicking the windows button is not bringing up the usual context menu. But well, at least the rest is working.

--> Switching to stable windows builds only now, to avoid this in the future. :)

**Comment by StealUrKill** [comment]
Brand new install on brand new hardware. Issue is the same as others except it does not even install all the way. 
![image](https://user-images.githubusercontent.com/35749471/142949034-60228482-043f-4d15-8cd2-0ffd1431ae87.png)

**Comment by djc5166** [comment]
Today's update broke EP for me, same issues as in this thread. v22000.348.

Not on any preview or beta channels.

**Comment by drixtol** [comment]
ditto @djc5166  - todays patch to .348 is the stable? patch i received.

**Comment by BraINstinct0** [comment]
@djc5166 @drixtol @valinet The current stable channel version is, according to [MS docs](https://docs.microsoft.com/windows/release-health/windows11-release-information#windows-11-current-versions), 22000.348. This means that having 'this will only work for stable versions' warning in README will be of no use.
Why, why are you not just releasing dbg symbols on time, MS......

**Comment by donsenilo1968** [comment]
The registry-hack works for me except right-click on the windows startbutton don't work anymore.
I'm not in the insider program or beta or whatever.

**Comment by RobertSpir** [comment]
In this case the dll files are almost identical to 346 versions, so just updating hashes to match the new versions is ok and everything works, but in general this will not work every time, especially when there will be more changes to the files

**Comment by Spearow2** [comment]
actually just used the reg edit to change to .348 and worked with it, not to mention it somehow fixed start11 from a non-working situation

**Comment by AnshulJ999** [comment]
I'm a noob and I'm confused - is there any fix here or do we just wait for EP to get updated? Losing productivity here cos the W11 taskbar is terrible :/

**Comment by Spearow2** [comment]
> 

check @Laserstrahl  reply above, reg edit to change the version, please note though that it seems that it doesn't work for everyone, for whatever reason

**Comment by BraINstinct0** [comment]
@AnshulJ999 No, this is not a problem of ExplorerPatcher - rather, it will be solved (and probably only) when Microsoft releases the debug symbols for 348.

**Comment by valinet** [comment]
> > 
> 
> check @Laserstrahl reply above, reg edit to change the version, please note though that it seems that it doesn't work for everyone, for whatever reason

It's not "whatever reason", it's well explained above in my reply.

**Comment by Spearow2** [comment]
> > > 
> > 
> > 
> > check @Laserstrahl reply above, reg edit to change the version, please note though that it seems that it doesn't work for everyone, for whatever reason
> 
> It's not "whatever reason", it's well explained above in my reply.

true true, bad choice of words, as you said it is rather the coincidence whether the data was saved or not.

**Comment by udeshplus** [comment]
Follow the below steps to refrain from installing the build 348 update; this works for people who got their version updated due to the windows insider program. 

- Go to Windows Insider Program -> Stop getting preview builds and turn on Unenroll this device
- Go to uninstall updates and remove "2021-11 Cumulative Update for Windows 11 for x64-based System (KB5007262)."
- Restart 
- Then Windows update again will install that update automatically. But it's okay let it install again. Once it's done, restart again.
- Now, you can see that you have successfully unenrolled from the insider program.
- Now go to windows updates and remove this update again "2021-11 Cumulative Update for Windows 11 for x64-based System (KB5007262)."
- Restart and install the ExplorerPatcher
- Now windows update wont automatically install the update. it will be listed as an optional update.
- See the screenshot. This is how I see the update.

![ooooooooooooooopppp](https://user-images.githubusercontent.com/30784707/143006757-9e7e7343-6d68-44a6-8e7e-dbccea42d66a.png)

**Comment by valinet** [comment]
Also, from here: https://www.howtogeek.com/398226/now-windows-10-has-c-b-and-d-updates.-what-is-microsoft-smoking/

> ## “C” and “D” Updates: “Optional” Preview Updates
>
> “C” and “D” updates are released in the third and fourth weeks of the month, respectively. These do not include any new security updates.
>
> These updates just include new bug fixes and improvements for other non-security issues. Microsoft says C and D updates are “optional,” and Windows Update won’t automatically install them on your PC.

So, again, I don't see the problem. Personally, I have Windows Update turned off entirely. I can't and don't want to keep up to date with their continuous update madness. Updating once a month when the real updates are released anyway is more than enough, I think, so I just manually check and install the patches around Patch Tuesday every month, so the computer does not "restart on its own", "starts using the CPU 100% on its own all of a sudden" etc, things Microsoft is famous in the past couple of years for when it comes to Windows Update and Windows Defender. "C" and "D" releases are just another name for yet another wave of beta testing they do before Patch Tuesday, testing the stuff on a smaller subset of users.

As mentioned by Microsoft itself, this 22000.348 preview update did not just install by itself (or if it did, it's Microsoft's problem again, but idk, I personally haven't experienced that). Here's one of my test VMs, it shows like this, same as @udeshplus:

![image](https://user-images.githubusercontent.com/6503598/143007931-955e0f4c-d88e-482c-97f8-4ab646674393.png)

You have to press that button for it to install. It's another story if you use Windows Insider. I honestly do not understand why you enroll in Windows Insider and then want to use a patcher to undo most of the changes in Insider, what's the point...? That's the latest shit in Windows, don't want that, then stay on the regular releases at least. I don't have an army behind me that looks at all the 4-5 different releases that are current at any point in time and that fixes EP for each of those. Try to balance out the factors involved before clicking some button and making a decision, put away the shiny propaganda Microsoft puts out about this Insider Program non-sense when deciding to use that. And like, it's clear it's just a giant beta test program, remember how Windows Subsystem for Android was first available in regular Windows 11 and only after a week or so in the Dev channel, like, even Microsoft doesn't give too much damns about it.

Also, I don't have time every few days now to keep up with whatever madness Microsoft is up to. EP is provided as is, it works as is, it's generally tested on the few scenarios I test on, and needs certain capabilities to enable certain scenarios other tools preferred not to delve into. I try to keep it working on the current build which I daily drive. I won't invest energy fixing it elsewhere, especially on Insider Preview for example, as it doesn't make sense: that's a staging area for Microsoft's latest experiments, doing something there is almost pointless as there is no guarantee what you patch against there will ever see the light of day in a shipping product.

Also, for maximum stability and a proper OS, I recommend staying on Windows 10. I don't see the need for an update, especially considering that you're here. I "upgraded" to Windows 11 for the following reasons which are niche cases I very rarely use, so rare that I highly question why I still waste time on this release (Windows 10 2004 it's been pretty much the same for 2 years, so a perfectly stable release, same as the good old 1809 which is also rock solid):

* improvements in how applications on multiple monitors are handled (i.e. when undocking a laptop, apps minimize, and then when connecting back, they are restored to where hey originally were)
* ~snap groups (i.e. hover over the maximize button of any window and a suggestion pops up with different layouts in which you can arrange the windows on your screen)~ for me, useless in practice since I have very large monitors so I never use that anyway
* support for hard disk mounting in WSL2
* support for GUI apps in WSL2 using WSLg
* support for GPU compute applications in WSL2

I don't fundamentally care about rounded corners, Mica, GUI updates etc; they're nice to have stuff, but not essential. The WSL2 stuff I use, but very rarely as well, to the point where virtualizing the entire computer and running a proper OS underneath, like GNU/Linux, or just booting to a proper OS, like GNU/Linux, from time to time when this is need, makes much more sense to me. Windows 10 is fine, no one asked for this madness, everyone wanted some meaningful improvements in some pain point areas in Windows 10, not to fuck with everyone's workflow just to provide a macOS like dock, which, let's be serious: whoever wants to use macOS, will actually use the genuine thing, this imitation is no where near that level, so who is it for then? I assure you, the centered Windows 11 taskbar will stop no frustrated Wintel users from jumping to M1.

I am tired of this project and its raison d'etre and also have to work on some other stuff preponderantly lately. It's been nice (and a bit terrifying, seeing what they are up to) learning about the internals of the shell, but, as I said, I cannot dedicate all the time to working on this and especially clearing the mess Microsoft makes every few days. If the community wants to step in and alleviate some of the problems or whatever, be my guest, but besides a few isolated cases, I haven't really seen that happening too much when it comes to this project.

When the symbols for 348 become available, this will automagically start working just fine again.

**Comment by zeropxl2** [comment]
> UPD. Fixed somehow, changed the `OSBuild` version to `10.0.22000.348` by `Computer\HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\ExplorerPatcher` Totally not sure it's the right way to fix but works for me.

Wow, his saved me from getting really frustrated. I forgot to move the taskbar from the left to the bottom of the screen before updating Windows 11, leaving me with a blank taskbar that I couldn't move.
After I changed the OSBuild version in the registry to match the stable Windows 11 version, and restarting the system got me the good old taskbar back.
And I'm able to apply the ExplorerPatcher tweaks.  

Thanks @BlackRockSoul

**Comment by Silbian** [comment]
Just updated Win11 to 22000.348 and Explorer Patcher ( latest version 318.38.2 ) stopped working, as expected...
Changing the Registry key, as above mentioned, EP is worning again ! No side effects at the moment. Thanks for the tip.

**Comment by Sinkers1** [comment]
I looked here Computer\HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\ExplorerPatcher in the registry, but could see no reference to the OSBuild. What am I missing?

**Comment by plofhaan** [comment]
>  EP is provided as is, it works as is,

Could you at least give us the _option_ to temporary disable the specific EP functions that specifically depend on the symbols? The reason I ask is because at least I won't have to stare at a blank screen with a white mouse arrow in the middle. Thanks a million for your kind patience and understanding. :)

**Comment by plofhaan** [comment]
As for the question of why I need 22000.348 right now, it fixes many bugs including dwm.exe memory leak which when it flurbs it causes my CPU load to increase by 22% each time until I reboot again.

**Comment by SkyDrift** [comment]
so, at this point what I'm reading that this project is to be dead this early in to win 11 build.

**Comment by plofhaan** [comment]
> so, at this point what I'm reading that this project is to be dead this early in to win 11 build.

Exactly.

**Comment by BraINstinct0** [comment]
@plofhaan Do you mean you'd like to uninstall it?
(sorry for bad formatting in advance - I do not intend to write a fully-fledged guide, just some advice.)
That is,
1) if it failed to fetch symbols EP won't hook into explorer.exe - which means you'll at least get the win 11 taskbar.
2) if you meant to 'uninstall', but cannot find the file to use(running the installer again prompts you whether you'll uninstall EP) or open the settings, 
 2-1) try Ctrl-Alt-Del, open task manager, File->Run new task(or whatever similar)->explorer.exe->press Enter then you'll possibly get the Win11 UI for now. Then you can run the uninstaller. If not, see 2-2)
 2-2) Next one you can try is uninstalling EP from settings. The settings is basically `ms-settings:`. This can again be achieved by multiple ways:
 2-2-1) try opening CMD or Powershell(`pwsh` or `powershell` will probably both work. if you haven't heard about this, you can ignore this parentheses entirely). then, type `start ms-settings:` , then press Enter.
 2-2-2) If, by any means, you can open any Browsers or the File Explorer, type `ms-settings:` in the address bar and press enter. (not search taht term, but treat it as a URL or file path.)
3) If above doesn't work or doesn't match your needs please detail about it in this thread.

**Comment by BraINstinct0** [comment]
@SkyDrift Depends. Usually (and stated earlier by MS) the debug symbols are uploaded very soon after a new release. However there seems to be quite a delay these days. So, it's not gonna die if MS keeps its words, but apparently they're not so reliable. Besides, if you really want Win10 experience you can use Win10, why not?

**Comment by BraINstinct0** [comment]
Pushing the dev generally has minimal effect.
Pushing support($) to the dev has some effect.
Pushing commits(PRs) to the dev's repo or providing help to the community usually mean there will be progress.

I just prefer the latter.

**Comment by valinet** [comment]
Please try the latest pre-release (.7). I guarded off the functionality requiring symbol until a better solution is implemented; this means that it will work on untested builds, but the functionality requiring the stuff from the files for which symbols are required won't work. Mainly:
* `Win`+`X` menu
* Skinning/unskinning some menus
* Having the Start menu open to "All apps" by default

@plofhaan @SkyDrift It's more useful to have a constructing and understanding attitude as it's not my fault this situation is happening and by the way, I have no obligation to fix this, furthermore, I offered you all the means and tools to go fix it yourself however you want as well, so I am in no way the one unreasonable here, not to mention I try to help in most if not all the situations encountered here; I have other stuff to do sometimes so I can't always reply to your requests by the minute, excuse me for that. Also, excuse me that I laid out honest replies telling the situation as is rather than lie you with all sorts of stories like others do.

It's also advisable to try to find solutions to problems rather than try to hack an application that hacks `explorer.exe`, just saying...

Anyway, please test this version and if it works well, it will be published as a release.

**Comment by Anixx** [comment]
> Application starts with limited functionality on builds lacking hardcoded symbol information

This is very much appreciated improvement! Among other things it also allows to ship Win11 initial build with EP installed and ready for work.

**Comment by Anixx** [comment]
For some reason does not update on my machine though...

**Comment by Anixx** [comment]
The new feature "Taskbar autohide" does not work for me...

**Comment by valinet** [comment]
> For some reason does not update on my machine though...
> 
> P.S. Updated via manual downloading.

Explained multiple times how it works, the updater only pulls releases, not pre-releases.

**Comment by Anixx** [comment]
> Explained multiple times how it works, the updater only pulls releases, not pre-releases.

Thanks. I have just found a relevant discussion on the forum.

**Comment by valinet** [comment]
> The new feature "Taskbar autohide" does not work for me...

What does it do, minimize all windows?

**Comment by plofhaan** [comment]
@valinet Where did you see me claim that it is your fault? To be clear, the only reason why I repeated my suggestion to add the gracious fallback type of approach in EP was because I honestly though it was constructive feedback and that you might have missed it somehow. I am not asking for anything, as you don't owe me anything. I am here only to offer you my suggestions, and because I can always appreciate having a good laugh about the funny subject of "Which computer monkey is to blame for Windows 11?". ;P

**Comment by Anixx** [comment]
> What does it do, minimize all windows?

Nothing. There is no effect on double-clicking on taskbar...

**Comment by hubjective** [comment]
I just joined to say that @udeshplus gave a miracle solution in their comment above.  [https://github.com/valinet/ExplorerPatcher/issues/440#issuecomment-976374458](url)  **It appears I have successfully left the Insider channel completely**, without having to do a clean install. I recently spent a few hours trying to figure out how to do this without a nuke and pave, so I think the unenroll setting must be new. I really wish unenrolling had been an option before I got stuck on Win11.

**Comment by plofhaan** [comment]
@BraINstinct0 I had already uninstalled it the moment after I got the blank screen, but thanks anyway. `Ctrl` + `Shift` + `Esc` opens Task Manager, BTW. I also don't know why you need to go in `ms-settings:` because I simply opend an elevated cmd to rename the dxgi.dll and then I restarted explorer after that.

**Comment by valinet** [comment]
@plofhaan Down voting my post where I explained how things stand, as well as "Exactly" when replying to that "project is dead" message helped me form a certain opinion. I have not missed it, I thought of it from the beginning as well, I just don't like it in particular, because it is very confusing - I cannot wait for the obligatory "`Win`+`X` does not work" etc posts when this becomes a release. I did not anticipate this symbols issue to become such a big problem, Microsoft used to update them on a timely manner. The release notes clearly contain the builds on which this works, apparently that is not enough and all sorts of hacks are always needed. When something breaks because people hack up all kinds of "solutions" instead of reading up and understanding the root cause, they naturally come here and blame the software. I explain all of this and you down vote me. What impression should I then make...?

@Anixx For me, in the W11 classic VM, it minimizes all windows. It's because of 7+TT where I had an option set for double click. Now I see it already offered that, so probably shouldn't have included this at all, idk... With that disabled, it works. Check for any conflicting programs. Most of the recent options are the result of personal preferences of users without a clear direction for the program. If I knew 7+TT offered this already, I wouldn't have added it in probably... As I said, I don't want to duplicate functionality...

@hubjective Yeah, it's very good as I feel like a lot of people are in Insider just so that they could get Windows 11. Another idiotic Microsoft policy. Insider is staging code, it's pretty rough and problematic to daily drive. This fortunately works as the updates are only CUs, I don't think it would have worked previously, when Insider updates meant new builds, or at least you would have been stuck on a time bombed build with nowhere to go.

**Comment by valinet** [comment]
> @BraINstinct0 I had already uninstalled it the moment after I got the blank screen, but thanks anyway. `Ctrl` + `Shift` + `Esc` opens Task Manager, BTW. I also don't know why you need to go in `ms-settings:` because I simply opend an elevated cmd to rename the dxgi.dll and then I restarted explorer after that.

He's just recommending the *proper* way to uninstall. The app got an installer/uninstaller a few releases ago, he just recommended using that via "Apps and features" in `ms-settings:`.

**Comment by plofhaan** [comment]
@valinet Sorry, but you said it yourself that you were basically done with the project. Whether people prefer getting a bank screen over missing various stuff like the WinX menu, skinned flyouts and the like, I'll leave that to the people to decide. Thank you.

**Comment by hubjective** [comment]
@valinet PP donation on its way. I couldn't use Win11 for even a day without your app.

I was, in fact, stuck with a Win10 version that was about to expire and that's why I moved to Win11. I had tried to do a clean install to get out of the beta program back when I made this computer my main one, but the beta version had overwritten my recovery partition and no matter what I tried, I couldn't get UEFI to boot anything that would clean install a non-beta Windows.

**Comment by StealUrKill** [comment]
I think he has the newer one which does this.

"Running ep-setup.exe again while EP is already installed will now update the program to the latest version. To uninstall, as the previous behavior did, run ep_setup.exe /uninstall"


He can run this is a batch file as well.

`
@echo off

taskkill /f /im explorer.exe
echo.
if exist "C:\Program Files\ExplorerPatcher\ep_setup.exe" "C:\Program Files\ExplorerPatcher\ep_setup.exe" /uninstall
echo.
if exist %windir%\dxgi.dll del %windir%\dxgi.dll /f /q
echo.
if exist %windir%\dxgi.prev del %windir%\dxgi.prev /f /q
echo.
start explorer.exe
`


> @plofhaan Do you mean you'd like to uninstall it? (sorry for bad formatting in advance - I do not intend to write a fully-fledged guide, just some advice.) That is,
> 
> 1. if it failed to fetch symbols EP won't hook into explorer.exe - which means you'll at least get the win 11 taskbar.
> 2. if you meant to 'uninstall', but cannot find the file to use(running the installer again prompts you whether you'll uninstall EP) or open the settings,
>    2-1) try Ctrl-Alt-Del, open task manager, File->Run new task(or whatever similar)->explorer.exe->press Enter then you'll possibly get the Win11 UI for now. Then you can run the uninstaller. If not, see 2-2)
>    2-2) Next one you can try is uninstalling EP from settings. The settings is basically `ms-settings:`. This can again be achieved by multiple ways:
>    2-2-1) try opening CMD or Powershell(`pwsh` or `powershell` will probably both work. if you haven't heard about this, you can ignore this parentheses entirely). then, type `start ms-settings:` , then press Enter.
>    2-2-2) If, by any means, you can open any Browsers or the File Explorer, type `ms-settings:` in the address bar and press enter. (not search taht term, but treat it as a URL or file path.)
> 3. If above doesn't work or doesn't match your needs please detail about it in this thread.
> 
> Disclaimer: I too, have uninstalled it for now - I use WSL2, WSA, etc. heavily, so am forced to use Windows 11 for now. This project is Opensource, if you need parts of it you may freely fork&edit it. (Blame MS, for its poor UI decision)



> @plofhaan Do you mean you'd like to uninstall it? (sorry for bad formatting in advance - I do not intend to write a fully-fledged guide, just some advice.) That is,
> 
> 1. if it failed to fetch symbols EP won't hook into explorer.exe - which means you'll at least get the win 11 taskbar.
> 2. if you meant to 'uninstall', but cannot find the file to use(running the installer again prompts you whether you'll uninstall EP) or open the settings,
>    2-1) try Ctrl-Alt-Del, open task manager, File->Run new task(or whatever similar)->explorer.exe->press Enter then you'll possibly get the Win11 UI for now. Then you can run the uninstaller. If not, see 2-2)
>    2-2) Next one you can try is uninstalling EP from settings. The settings is basically `ms-settings:`. This can again be achieved by multiple ways:
>    2-2-1) try opening CMD or Powershell(`pwsh` or `powershell` will probably both work. if you haven't heard about this, you can ignore this parentheses entirely). then, type `start ms-settings:` , then press Enter.
>    2-2-2) If, by any means, you can open any Browsers or the File Explorer, type `ms-settings:` in the address bar and press enter. (not search taht term, but treat it as a URL or file path.)
> 3. If above doesn't work or doesn't match your needs please detail about it in this thread.
> 
> Disclaimer: I too, have uninstalled it for now - I use WSL2, WSA, etc. heavily, so am forced to use Windows 11 for now. This project is Opensource, if you need parts of it you may freely fork&edit it. (Blame MS, for its poor UI decision)

**Comment by plofhaan** [comment]
> > @BraINstinct0 I had already uninstalled it the moment after I got the blank screen, but thanks anyway. `Ctrl` + `Shift` + `Esc` opens Task Manager, BTW. I also don't know why you need to go in `ms-settings:` because I simply opend an elevated cmd to rename the dxgi.dll and then I restarted explorer after that.
> 
> He's just recommending the _proper_ way to uninstall. The app got an installer/uninstaller a few releases ago, he just recommended using that via "Apps and features" in `ms-settings:`.

Already did that by typing appwiz in the search box of Classic Start Menu after I had restarted explorer.

**Comment by valinet** [comment]
> @valinet Sorry, but you said it yourself that you were basically done with the project. Whether people prefer getting a bank screen over missing various stuff like the WinX menu, skinned flyouts and the like, I'll leave that to the people to decide. Thank you.

I don't understand the need to constantly attack me. It's not like I want to cause people headaches either. I thought it was pretty clear since the time I released this tool anyway that I generally try to help. Of course, from your position, it's very easy to say all the populist stuff in the world, it's not you that stay awake all night long trying to help and fix it when someone's got a problem or something's broken or devise the best solution to tackle a problem. You just care about your interest on the moment and spreading some hate messages all around the places. I am pro identifying real solutions, this is still a band aid, but yeah, it works for now and I implemented it as soon as I got the opportunity, even though I don't like it. What did I do wrong still? I called for a discussion on how to solve this in general, you don't care, I called for feedback on whether .7 works, you did not bother, it was more interesting to hate on me or @BraINstinct0 than to help construct.

I haven't said "I am basically done with the project". I said I am tired of it, and that's something else, and it's still true. Still, that's the reason for down voting and immediately trash talking?

**Comment by plofhaan** [comment]
@valinet Where did I attack you? Like I said, I ask nothing, and, you owe me nothing. If my suggestions are not welcomes here or worse, you consider them to be attacks, then I think this says more about your own personal attitude than it says about me. I only downvote a message if I don't agree with it, if I have a different opinion. There's absolutely no hate involved with that. Have a nice day!

**Comment by StealUrKill** [comment]
So @valinet I see where you say if this link [http://msdl.microsoft.com/download/symbols/twinui.pcshell.pdb/73402D8892AF2DA2639F29A5D2F483911/twinui.pcshell.pdb](url) serves you a file it will work.

Does the program download that file during installation? 

My other computer that works due to it being an older winver does not have that file or folder located in \User\"NAMEHERE"\\AppData\Roaming\

ExplorerPatcher Directory is completely missing on the working machine. 

Just trying to understand the operation is all.

Thanks for your time.


> @plofhaan Down voting my post where I explained how things stand, as well as "Exactly" when replying to that "project is dead" message helped me form a certain opinion. I have not missed it, I thought of it from the beginning as well, I just don't like it in particular, because it is very confusing - I cannot wait for the obligatory "`Win`+`X` does not work" etc posts when this becomes a release. I did not anticipate this symbols issue to become such a big problem, Microsoft used to update them on a timely manner. The release notes clearly contain the builds on which this works, apparently that is not enough and all sorts of hacks are always needed. When something breaks because people hack up all kinds of "solutions" instead of reading up and understanding the root cause, they naturally come here and blame the software. I explain all of this and you down vote me. What impression should I then make...?
> 
> @Anixx For me, in the W11 classic VM, it minimizes all windows. It's because of 7+TT where I had an option set for double click. Now I see it already offered that, so probably shouldn't have included this at all, idk... With that disabled, it works. Check for any conflicting programs. Most of the recent options are the result of personal preferences of users without a clear direction for the program. If I knew 7+TT offered this already, I wouldn't have added it in probably... As I said, I don't want to duplicate functionality...
> 
> @hubjective Yeah, it's very good as I feel like a lot of people are in Insider just so that they could get Windows 11. Another idiotic Microsoft policy. Insider is staging code, it's pretty rough and problematic to daily drive. This fortunately works as the updates are only CUs, I don't think it would have worked previously, when Insider updates meant new builds, or at least you would have been stuck on a time bombed build with nowhere to go.

**Comment by valinet** [comment]
@StealUrKill That link is for `twinui.pcshell.dll` from another build. I said that in the thread where this fiasco happened yet again with some previous pre-release build of Windows. For this 348 build, the location of the symbol is something else, mainly that hash in the URL differes, that's how it knows which symbol to pick when they all have the same name, that's the hash of the file for which that symbol is valid.

The directory in `%APPDATA%` is created under certain circumstances only. For some builds, the addresses the program would manually obtain from the symbol file are hardcoded in the source code, so it does not have to do the process manually and so it does not create that folder to store the temporary downloads.

Symbols are downloaded (or rather, were, now that's disabled by default - I mean in 38.7) automatically on the first run of the application, if hard coded values cannot be already found in the program itself for the files on your system.

**Comment by SpIdErPiGgY1984** [comment]
I would be very sorry if you stopped this project. I really need the features of this patch. I have left the "beta" program of Windows, and installed the stable Windows 11 version. Because I understand that it really takes a lot of time to make these things, and I certainly don't want to make it more difficult for you. So, thanks for this super patch @valinet  ❤️❤️❤️

**Comment by StealUrKill** [comment]
@valinet Sorry for the questions. Looking over the code and all trying to learn what it is doing is new to me. I assume the patcher finds the hash itself? And if that url with the hash it decoded is not present, then that is the issue correct?

I am used to Android source code lol.

Just to confirm its the dll or the pdb? The dll is present on the machine.

![image](https://user-images.githubusercontent.com/35749471/143067062-2199705e-0a92-4507-85ba-babe40487f77.png)

**Comment by valinet** [comment]
@StealUrKill I don't understand the question. The URL at which the symbol file (PDB) for a DLL from the Windows OS can be found at some well documented place in the DLL itself, which is in the PE format. Once EP gets that URL, it tries to download the file from there, which is the required symbol file. If that URL does not work, it means Microsoft hasn't published the symbols for that file, yet or they may never do... Symbols are never shipped with Windows, since it would make the OS 100GB at least, if not more =))) and it's pretty hard to do that anyway from other considerations, so no, you will not find any PDB on your computer by default. EP places the file in that folder temporarily to extract data from it and then it is done with it. Was this what you were asking?

**Comment by StealUrKill** [comment]
@valinet ok yes that makes sense now. I am wondering through the dll using HxD and x64dbg lol.

Assuming the extracted data is then sent to EP in a file or directory for use?

**Comment by StealUrKill** [comment]
nevermind i think i see the iconcache_XX.db and where the info is.

unless im wrong again lol

**Comment by valinet** [comment]
@StealUrKill what are you doing exactly? The PE format is described on MSDN clearly. The link to the PDB file is in the file itself, but it's not literal, it's constructed from various things in there. I don't remember exactly, I coded this 1 year ago or more, look on the source code: https://github.com/valinet/libvalinet/blob/ecbd725891f24bbdb06912f2caa98c82f0e94456/valinet/pdb/pdb.h#L420. If you want to get that, drop the DLL in a tool like [PDBDownloader](https://github.com/rajkumar-rangaraj/PDB-Downloader) and it will show you the URL.

**Comment by StealUrKill** [comment]
@valinet I seen what i was looking for. I appreciate all the info and help. Also appreciate the link for the downloader.

**Comment by StealUrKill** [comment]
@valinet i managed to find another program that will spit out the url via command line and not error out like PDBDownloader you linked.

The URL seems to be
http://msdl.microsoft.com/download/symbols/twinui.pcshell.pdb/19708DC7FB9C2034C85A07FE76E823791/twinui.pcshell.pdb

[CMD LINE PDBDownloader](https://github.com/Biswa96/PDBDownloader)

**Comment by valinet** [comment]
Nice, yet "error out"... the file is unavailable, the URL is probably that indeed, but still nothing shows up when you visit it, so Microsoft still hasn't published anything. They seem to only support development on the stable build and on dev Insider Preview, as symbols are basically published only for that.

**Comment by StealUrKill** [comment]
@valinet yes "error out" was a bad choice of words. Yes it seems the feature experience pack for the stable build has not been pushed yet. They just seem to be late hopefully. Thanks again for all the info. Definitely helps me when trouble shooting.

**Comment by 0xF5F5F5** [comment]
Guess I shouldn't have installed that optional Windows update :^)
Was wondering "why hasn't Microsoft published symbols for a stable release?", but now I remembered I manually checked for Windows updates and accepted a system update (22000.348) that is currently optional.

Thank you to all contributors. I love ExplorerPatcher! I'm impressed with what this tool achieves in re-surfacing and re-implementing Windows 10 behavior, and I've regained some of my sanity because of it!

**Comment by HoRnEyDvL** [comment]
I had the same issue with 348 however im not part of any insider / preview chan, I did however  ran an optional updated.

I ended up just uninstalling update KB5007262 & now taskbar is working again.

Temp fix until a more permanent solution comes out but i'm happy for now.

Thanks for the amazing work you done on this tool Valinet

**Comment by StealUrKill** [comment]
@HoRnEyDvL Pray Tell how you uninstalled it? From there notes alone you can't since its a SSU. I manually removed via DISM and it totally messed everything up.

`If you want to remove the LCU

To remove the LCU after installing the combined SSU and LCU package, use the DISM/Remove-Package command line option with the LCU package name as the argument. You can find the package name by using this command: DISM /online /get-packages.

Running Windows Update Standalone Installer (wusa.exe) with the /uninstall switch on the combined package will not work because the combined package contains the SSU. You cannot remove the SSU from the system after installation.`

**Comment by valinet** [comment]
@StealUrKill Probably using this...?

![image](https://user-images.githubusercontent.com/6503598/143145380-a5359c52-0def-4f9d-992c-c426b05b6aa0.png)

**Comment by StealUrKill** [comment]
> @StealUrKill Probably using this...?
> 
> ![image](https://user-images.githubusercontent.com/6503598/143145380-a5359c52-0def-4f9d-992c-c426b05b6aa0.png)
> 

@valinet it won't let me from there. Uninstall in greyed out .

**Comment by valinet** [comment]
Idk, it works just fine over here. I haven't seen things there that are uninstallable, but yeah, I am not saying weird things can't happen on Windows...

**Comment by henjue** [comment]
@valinet I don't know what happened. I specially reinstalled the system (stable version). You don't have to pay attention to some people They don't do anything and always point! But I hope you stick to it. Thank you very much!

**Comment by zoghal** [comment]
> Nice, yet "error out"... the file is unavailable, the URL is probably that indeed, but still nothing shows up when you visit it, so Microsoft still hasn't published anything. They seem to only support development on the stable build and on dev Insider Preview, as symbols are basically published only for that.

I just now checked again and saw that the file can be downloaded. :)
http://msdl.microsoft.com/download/symbols/twinui.pcshell.pdb/19708DC7FB9C2034C85A07FE76E823791/twinui.pcshell.pdb

**Comment by matzoshaina** [comment]
Confirming symbol files are now available for build 348, explorerpatcher  working fine with retention of toolbars as well as notification icons. Just an example of delayed release of symbol files with a new build that we have seen before

**Comment by Galixte** [comment]
@frankbons please close this issue is solved. ;)

**Comment by BraINstinct0** [comment]
Alternative: leave this open, but rename title to "Feature X of ExplorerPatcher doesn't work anymore after Update to Windows Build YYYYY". This kind of situation might(for me: inevitably) occur again, but users who look up issues(if any) don't see closed issues often.

**Comment by frankbons** [comment]
Solved.

**Comment by valinet** [comment]
Please refer to this material I have published as well, it mainly describes the behavior which will be introduced in the next version which will be published soon: https://github.com/valinet/ExplorerPatcher/wiki/Symbols

Thanks
